app/controllers/api/v1/messages_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Api
+  module V1
+    class MessagesController < ApplicationController
+      before_action :authenticate_user!
+      before_action :set_conversation, only: [:create]
+      before_action :set_conversation_messages, only: [:index]
+
+      def create
```

Comment:
You should use [Action Cable](https://guides.rubyonrails.org/action_cable_overview.html) for this, and remove the create endpoint. Maybe this tutorial can help you:

https://medium.com/@mreigen/how-to-use-rails-5-action-cable-to-build-a-chat-system-with-multiple-private-rooms-with-multiple-5bf82d98a5aa